src/contract_wrappers/exchange_wrapper.ts

```diff
@@ -150,13 +150,18 @@ export class ExchangeWrapper extends ContractWrapper {
         assert.doesConformToSchema('signedOrder', signedOrder, schemas.signedOrderSchema);
         assert.isBigNumber('fillTakerTokenAmount', fillTakerTokenAmount);
         assert.isBoolean('shouldThrowOnInsufficientBalanceOrAllowance', shouldThrowOnInsufficientBalanceOrAllowance);
+        // this is already being done in validateFillOrderThrowIfInvalidAsync
         await assert.isSenderAddressAsync('takerAddress', takerAddress, this._web3Wrapper);
 
+        // we can chain this with gas cost estimate and parallelize the resulting promise with validateFillOrderThrowIfInvalidAsync below
         const exchangeInstance = await this._getExchangeContractAsync();
+
+        // we can do this optionally
         await this.validateFillOrderThrowIfInvalidAsync(signedOrder, fillTakerTokenAmount, takerAddress);
 
         const [orderAddresses, orderValues] = ExchangeWrapper._getOrderAddressesAndValues(signedOrder);
 
+        // can we do this optionally as well? we can have the caller pass in custom gas and gasPrice
```

Comment:
Mh... don't think we want them to supply a gas amount.

Reply:
there is some possible optimization here, i suspect the estimated gas cost will be very similar between different fills

Reply:
I agree that this `estimateGas` call isn't optimal. I think we should allow passing in the gas limit/price and fallback to `estimateGas`. Theres an generally an upper limit to the amount of gas a fill require, but it could be different with more complex tokens (e.g MiniMe tokens).